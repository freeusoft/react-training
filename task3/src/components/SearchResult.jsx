import React, { Component } from 'react'

class SearchResult extends Component {
  render () {
    return (
      <div className='content'>{`Content component with: ${this.props.content} (Functional component)`}</div>
    )
  }
}

export default SearchResult
