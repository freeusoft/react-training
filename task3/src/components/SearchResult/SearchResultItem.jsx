import React, { Component } from 'react'
import { Link } from 'react-router-dom'

export default class SearchResultItems extends Component {
  constructor (props) {
    super(props)

    this.state = {
      film: props.film
    }
  }

  render () {
    return (
      <Link to={'/result/' + this.props.film.id}>
        <div className='search-result-item card'>
          <img className='search-result-item-poster card-img-top' src={this.props.film.poster} />
          <div className='search-result-item-description'>
            <span className='search-result-item-name card-title'>{this.props.film.name}</span>
            <span className='search-result-item-year'>{this.props.film.year}</span>
          </div>
          <div className='search-result-item-genre card-text'>{this.props.film.genre}</div>
        </div>
      </Link>
    )
  }
}
