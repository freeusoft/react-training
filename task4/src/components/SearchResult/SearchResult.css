a.search-result-item-link:hover {
    text-decoration: none;
}

.search-result-items {
    padding-left: 40px;
    padding-right: 25px;
    margin-top: 40px;
}

.search-result-item {
    width: 15rem;
    margin: 25px;
    background-color: #ecf0f1!important;
}

.search-result-item:hover {
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.5);
}

.search-result-item-poster {
    padding-bottom: 10px;
}

.search-result-item-description {
    padding-left: 10px;
    padding-right: 10px;
    text-decoration: none;
}

.search-result-item-name {
    text-transform: uppercase;
    font-weight: bold;
    color: rgb(34, 34, 34);
}

.search-result-item-year {
    font-size: 8pt;
    border-style: solid;
    border-radius: 2pt;
    border-width: 1px;
    width: 30px;
    text-align: center;
    float: right;
    color: rgb(34, 34, 34);
}

.search-result-item-genre {
    font-size: 8pt;
    padding-left: 10px;
    padding-bottom: 10px;
    color: rgb(87, 87, 87);
}

.search-result {
    min-height: 450px;
}

.search-result-not-found {
    font-size: 35pt;
    padding-top: 180px;
    text-align: center;
    color: darkgray;
    min-height: 450px;
}