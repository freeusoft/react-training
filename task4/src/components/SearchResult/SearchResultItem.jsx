import React, { Component } from 'react'
import { Link } from 'react-router-dom'

export default class SearchResultItems extends Component {
  constructor (props) {
    super(props)

    this.state = {
      film: props.film
    }
  }

  itemClickHandler () {
    console.log('click')
    window.scrollTo(0, 0)
  }

  render () {
    return (
      <Link to={'/result/' + this.props.film.id} onClick={this.itemClickHandler} className='search-result-item-link'>
        <div className='search-result-item card'>
          <img className='search-result-item-poster card-img-top' src={this.props.film.poster} />
          <div className='search-result-item-description'>
            <span className='search-result-item-name card-title'>{this.props.film.name}</span>
            <span className='search-result-item-year'>{this.props.film.year}</span>
          </div>
          <div className='search-result-item-genre card-text'>{this.props.film.genre}</div>
        </div>
      </Link>
    )
  }
}
