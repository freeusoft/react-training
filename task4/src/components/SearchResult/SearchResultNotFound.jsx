import React, { Component } from 'react'

export default class SearchResultNotFound extends Component {
  render () {
    return (
      <div className='search-result-not-found'>No films found</div>
    )
  }
}
