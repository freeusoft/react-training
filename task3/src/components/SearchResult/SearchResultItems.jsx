import React, { Component } from 'react'
import SearchResultItem from './SearchResultItem'

export default class SearchResultItems extends Component {
  constructor (props) {
    super(props)

    this.state = {
      filmsStub: [
        {id: 1, name: 'Kill bill', year: 2004, genre: 'Action & Adventure', director: 'Quentin Tarantino', poster: 'https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg', rating: 10},
        {id: 2, name: 'Kill bill2', year: 2004, genre: 'Action & Adventure', director: 'Quentin Tarantino', poster: 'https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg', rating: 10},
        {id: 3, name: 'Kill bill3', year: 2004, genre: 'Action & Adventure', director: 'Quentin Tarantino', poster: 'https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg', rating: 10},
        {id: 4, name: 'Kill bill4', year: 2004, genre: 'Action & Adventure', director: 'Quentin Tarantino', poster: 'https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg', rating: 10},
        {id: 5, name: 'Kill bill5', year: 2004, genre: 'Action & Adventure', director: 'Quentin Tarantino', poster: 'https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg', rating: 10},
        {id: 6, name: 'Kill bill6', year: 2004, genre: 'Action & Adventure', director: 'Quentin Tarantino', poster: 'https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg', rating: 10},
        {id: 7, name: 'Kill bill7', year: 2004, genre: 'Action & Adventure', director: 'Quentin Tarantino', poster: 'https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg', rating: 10}
      ]
    }
  }

  render () {
    return (
      <div className='search-result-items row'>
        {
          this.state.filmsStub.map((elm, i) =>
            <SearchResultItem key={i} film={elm} />
          )
        }
      </div>

    )
  }
}
