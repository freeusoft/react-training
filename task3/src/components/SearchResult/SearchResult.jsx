import React, { Component } from 'react'
import SearchResultHeader from '../SearchResultHeader/SearchResultHeader'
import SearchResultItems from './SearchResultItems'
import SearchResultNotFound from './SearchResultNotFound'

class SearchResult extends Component {
  componentWillReceiveProps (props) {
    if (this.props.not_found) {
      this.setState({'not_found': true})
    }
  }

  render () {
    const notFound = false
    return (
      <div className='search-result'>
        { notFound
          ? <SearchResultNotFound /> : (
            <div>
              <SearchResultHeader {...this.props} />
              <SearchResultItems />
            </div>)
        }
      </div>
    )
  }
}

export default SearchResult
