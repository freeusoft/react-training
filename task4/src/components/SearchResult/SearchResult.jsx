import React, { Component } from 'react'
import SearchResultHeader from '../SearchResultHeader/SearchResultHeader'
import SearchResultItems from './SearchResultItems'
import SearchResultNotFound from './SearchResultNotFound'
import './SearchResult.css'

class SearchResult extends Component {
  render () {
    return (
      <div className='search-result'>
        { this.props.not_found
          ? <SearchResultNotFound /> : (
            <div>
              <SearchResultHeader {...this.props} />
              <SearchResultItems />
            </div>)
        }
      </div>
    )
  }
}

export default SearchResult
