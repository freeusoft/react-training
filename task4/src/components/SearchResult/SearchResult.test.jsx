import React from 'react'
import SearchResult from './SearchResult'
import SearchResultItem from './SearchResultItem'
import SearchResultItems from './SearchResultItems'
import SearchResultNotFound from './SearchResultNotFound'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('SearchResult component suite', ()=>{
  test('SearchResult component text equal', () => {
    const wrapper = shallow(
      <SearchResult />
    )
    const div = wrapper.find('.search-result')
    expect(div.text()).toBe('<SearchResultHeader /><SearchResultItems />')
  })

  test('Not found', () => {
    const wrapper = shallow(
      <SearchResult not_found='true'/>
    )
    const div = wrapper.find('.search-result')
    expect(div.text()).toBe('<SearchResultNotFound />')
  })

  test('Not found with ComponentWillReceiveProps', () => {
    const wrapper = shallow(
      <SearchResult />
    )
    wrapper.setProps({not_found: true})
    const div = wrapper.find('.search-result')
    expect(div.text()).toBe('<SearchResultNotFound />')
  })

})

describe('SearchResultItem component test suite', ()=>{
  const film = {id: 1, name: 'Kill bill', year: 2004, genre: 'Action & Adventure', director: 'Quentin Tarantino', poster: 'https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg', rating: 10}
  
  test('SearchResultItem renders correctly', () => {
    const tree = renderer.create(
      <SearchResultItem film={film}/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Item click', () => {
    const wrapper = shallow(
      <SearchResultItem film={film}/>
    )
    const scrollToMock = jest.fn()
    global.scrollTo = scrollToMock

    wrapper.simulate('click')
    expect(scrollToMock.mock.calls.length).toBe(1)
    expect(wrapper.props().href).toEqual('/result/' + film.id)
  })
})

describe('SearchResultItem component test suite', ()=>{
  
  test('SearchResultItems renders correctly', () => {
    const tree = renderer.create(
      <SearchResultItems />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})

describe('Header component test suite', ()=>{
  test('Header title should have text', () => {
    const wrapper = shallow(
      <SearchResultNotFound />
    )
    
    expect(wrapper.text()).toContain('No films found')
  })

})
