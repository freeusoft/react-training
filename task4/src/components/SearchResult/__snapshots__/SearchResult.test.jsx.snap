// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultItem component test suite SearchResultItem renders correctly 1`] = `
<a
  className="search-result-item-link"
  href={undefined}
  onClick={[Function]}
>
  <div
    className="search-result-item card"
  >
    <img
      className="search-result-item-poster card-img-top"
      src="https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg"
    />
    <div
      className="search-result-item-description"
    >
      <span
        className="search-result-item-name card-title"
      >
        Kill bill
      </span>
      <span
        className="search-result-item-year"
      >
        2004
      </span>
    </div>
    <div
      className="search-result-item-genre card-text"
    >
      Action & Adventure
    </div>
  </div>
</a>
`;

exports[`SearchResultItems component test suite SearchResultItems renders correctly 1`] = `
<div
  className="search-result-items row"
>
  <a
    className="search-result-item-link"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="search-result-item card"
    >
      <img
        className="search-result-item-poster card-img-top"
        src="https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg"
      />
      <div
        className="search-result-item-description"
      >
        <span
          className="search-result-item-name card-title"
        >
          Kill bill
        </span>
        <span
          className="search-result-item-year"
        >
          2004
        </span>
      </div>
      <div
        className="search-result-item-genre card-text"
      >
        Action & Adventure
      </div>
    </div>
  </a>
  <a
    className="search-result-item-link"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="search-result-item card"
    >
      <img
        className="search-result-item-poster card-img-top"
        src="https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg"
      />
      <div
        className="search-result-item-description"
      >
        <span
          className="search-result-item-name card-title"
        >
          Kill bill2
        </span>
        <span
          className="search-result-item-year"
        >
          2004
        </span>
      </div>
      <div
        className="search-result-item-genre card-text"
      >
        Action & Adventure
      </div>
    </div>
  </a>
  <a
    className="search-result-item-link"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="search-result-item card"
    >
      <img
        className="search-result-item-poster card-img-top"
        src="https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg"
      />
      <div
        className="search-result-item-description"
      >
        <span
          className="search-result-item-name card-title"
        >
          Kill bill3
        </span>
        <span
          className="search-result-item-year"
        >
          2004
        </span>
      </div>
      <div
        className="search-result-item-genre card-text"
      >
        Action & Adventure
      </div>
    </div>
  </a>
  <a
    className="search-result-item-link"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="search-result-item card"
    >
      <img
        className="search-result-item-poster card-img-top"
        src="https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg"
      />
      <div
        className="search-result-item-description"
      >
        <span
          className="search-result-item-name card-title"
        >
          Kill bill4
        </span>
        <span
          className="search-result-item-year"
        >
          2004
        </span>
      </div>
      <div
        className="search-result-item-genre card-text"
      >
        Action & Adventure
      </div>
    </div>
  </a>
  <a
    className="search-result-item-link"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="search-result-item card"
    >
      <img
        className="search-result-item-poster card-img-top"
        src="https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg"
      />
      <div
        className="search-result-item-description"
      >
        <span
          className="search-result-item-name card-title"
        >
          Kill bill5
        </span>
        <span
          className="search-result-item-year"
        >
          2004
        </span>
      </div>
      <div
        className="search-result-item-genre card-text"
      >
        Action & Adventure
      </div>
    </div>
  </a>
  <a
    className="search-result-item-link"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="search-result-item card"
    >
      <img
        className="search-result-item-poster card-img-top"
        src="https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg"
      />
      <div
        className="search-result-item-description"
      >
        <span
          className="search-result-item-name card-title"
        >
          Kill bill6
        </span>
        <span
          className="search-result-item-year"
        >
          2004
        </span>
      </div>
      <div
        className="search-result-item-genre card-text"
      >
        Action & Adventure
      </div>
    </div>
  </a>
  <a
    className="search-result-item-link"
    href={undefined}
    onClick={[Function]}
  >
    <div
      className="search-result-item card"
    >
      <img
        className="search-result-item-poster card-img-top"
        src="https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg"
      />
      <div
        className="search-result-item-description"
      >
        <span
          className="search-result-item-name card-title"
        >
          Kill bill7
        </span>
        <span
          className="search-result-item-year"
        >
          2004
        </span>
      </div>
      <div
        className="search-result-item-genre card-text"
      >
        Action & Adventure
      </div>
    </div>
  </a>
</div>
`;
